File: wc.pl
#!/usr/bin/perl

@ARGV = ("-") unless @ARGV;

$count = 0;

print "Perl\n";
print "Processing $ARGV[0] ...\n";

open(TXT, $ARGV[0]) or die "Failed to open $ARGV[0]: $!\n";

%wc = ();

for $line (<TXT>) {
    $count++;

    # Remove \n$
    chomp($line);
    #if (length($line) == 0) {
    #    next;
    #}

    for $word (split(/ /, $line)) {
        $wc{$word}++;
    }
}

close(TXT);

print "Processed $count lines.\n";

for $i (sort { $wc{$b} <=> $wc{$a} } keys %wc) {
    printf("%10d: [%s]\n", $wc{$i}, $i);
}

File: wc.js
#!/usr/bin/env node

const fs = require("fs");
const readline = require("readline");

file = process.argv[2];

console.log("Javascript")
console.log("Processing %s ...", file)

let fp = readline.createInterface({
  input: fs.createReadStream(file),
  //output: process.stdout,
  console: false,
});

let count = 0
let wc = {}
fp.on("line", (line) => {
  count++
  line.split(" ").forEach((w) => {
    /*
    if (w.length == 0) {
      return;
    }
    */
    wc[w] ? wc[w]++ : wc[w] = 1;
  });
});

fp.on("close", () => {
  console.log("Processed %d lines.", count)

  let sorted = Object.keys(wc).sort((a, b) => wc[b] - wc[a]);

  sorted.forEach((w) => {
    // Note: Console log is very slow, even piped to > /dev/null.
    // console.log("%d: [%s]", wc[w], w);
  });
});
